pkg/solana/chainreader/event_read_binding.go

```diff
@@ -75,6 +78,46 @@ func (b *eventReadBinding) Decode(ctx context.Context, bts []byte, outVal any) e
 	return b.codec.Decode(ctx, bts, outVal, itemType)
 }
 
+func (b *eventReadBinding) GetLatestValue(ctx context.Context, params, returnVal any) error {
+	itemType := codec.WrapItemType(true, b.namespace, b.genericName)
+
+	pubKey, err := b.GetAddress(ctx, nil)
+	if err != nil {
+		return err
+	}
+
+	offChain, err := b.normalizeParams(params, itemType)
+	if err != nil {
+		return err
+	}
+
+	subkeyFilters, err := b.extractFilterSubkeys(offChain)
+	if err != nil {
+		return err
+	}
+
+	limiter := query.NewLimitAndSort(query.CountLimit(1), query.NewSortBySequence(query.Desc))
+	filter, err := logpoller.Where(
+		logpoller.NewAddressFilter(pubKey),
+		logpoller.NewEventSigFilter(b.eventSig[:]),
+		subkeyFilters,
+	)
+	if err != nil {
+		return err
+	}
+
+	logs, err := b.reader.FilteredLogs(ctx, filter, limiter, b.namespace+"-"+pubKey.String()+"-"+b.genericName)
```

Comment:
Not sure how many other places there are where we do this `b.namespace+"-"+pubKey.String()+"-"+b.genericName", but assuming this isn't the only place a helper for it might make sense

Reply:
Probably another thing that might go well in common since evm is very similar.